lib/boilerplate.js

```diff
@@ -51,6 +55,11 @@ const boilerplate = function (gulp, opts) {
 
   opts = _.merge({}, DEFAULT_OPTS, opts);
 
+  if (opts.typescript) {
+    // rewrite file matcher to find .ts
+    opts.files.push('*.ts', '**/*.ts', '!node_modules/**/*.ts');
```

Comment:
I'm not sure how this is "rewriting". If `opts.files` is already 
```js
['*.js', 'lib/**/*.js', 'test/**/*.js', '!gulpfile.js']
```
won't in now be 
```js
['*.js', 'lib/**/*.js', 'test/**/*.js', '!gulpfile.js', '*.ts', '**/*.ts', '!node_modules/**/*.ts']
```

Reply:
Yes, that's intentional. If we're using the typescript compiler, we're no longer using babel. The typescript compile can also transpile JS, so this enables us to keep a project working while migrating one file at a time from .js to .ts.